src/App.tsx

```diff
@@ -1,14 +1,66 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilteredFields } from './types/FilteredFields';
+
+type FilterObject = {
+  query: string;
+  filteredField?: FilteredFields | string;
+};
+
+function prepareString(str: string) {
+  return str.toLowerCase().trim();
+}
+
+const prepareTodos = (todos: Todo[], { query, filteredField }: FilterObject) => {
+  let todosCopy = [...todos];
+
+  if (query) {
+    const preparedQuery = prepareString(query);
+
+    todosCopy = todosCopy.filter(
+      todo => prepareString(todo.title).includes(preparedQuery),
+    );
+  }
+
+  if (filteredField !== FilteredFields.All) {
+    todosCopy = todosCopy.filter(todo => {
+      switch (filteredField) {
+        case FilteredFields.Active:
+          return !todo.completed;
+        case FilteredFields.Completed:
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+  }
+
+  return todosCopy;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [filteredField, setFilteredField] = useState<FilteredFields | string>(FilteredFields.All);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos().then(setTodos).finally(() => setLoading(false));
+  }, []);
+
+  const preparedTodos = prepareTodos(todos, { query, filteredField });
```

Comment:
You can wrap this function into useMemo to prevent function re-calls on each re-render